### PR TITLE
1388 docs update grid page

### LINKED ISSUE #1388: [Docs?]: update Grid page
### Latest version

- [X] I have checked the latest version

### Summary 💡

Update the next Grid page to use the new React API and page content.

### Motivation 🔦

Docs, Ref, Grid

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
closes #1388 

## What is the new behavior?
- Updates grid.mdx to use Grid component 
- Adds core-js required dep to build

### CODE REVIEW

File: website/docs/reference/components/grid.mdx [hunk 1]
```diff
@@ -21,94 +28,162 @@ import LiveExampleTabs from '@site/src/components/LiveExampleTabs/LiveExampleTab
...
-        ...styles.style,
-        backgroundColor: 'var(--ps-surface-strong)',
-        borderRadius: '6px',
-        height: '11rem',
-        marginBottom: '1rem',
-        padding: '1rem',
-      },
-    }
-  }
+Here is an example of how columns work with the Grid. Here we are creating a 12 column grid with a gap of 32px between each column.
```
Comment (Rykus0) [inline website/docs/reference/components/grid.mdx:38]:
```suggestion
Here is an example of how columns work with the Grid. Here we are creating a 12 column grid with a gap of 6px between each column.
```

Comment (caseybaggz) [reply website/docs/reference/components/grid.mdx:38]: @Rykus0 I need to fix a UI bug from the HS update. I'll update all of these so don't waste your time commenting. 😂

File: website/docs/reference/components/grid.mdx [hunk 2]
```diff
@@ -21,94 +28,162 @@ import LiveExampleTabs from '@site/src/components/LiveExampleTabs/LiveExampleTab
...
+Here is an example of how to span columns with the GridItem. Here we are creating a 12 column grid with a gap of 32px between each column. We are also spanning the first GridItem to take up 4 columns.
 
-```typescript
-getGridProps(options?: GridOptions): ReturnProps
+```jsx live
+function GridSpanExample() {
+  return (
+    <Grid cols="12" gap="6" rows="3rem">
+      <GridItem
+        colSpan="1 / span 4"
```
Comment (Rykus0) [inline website/docs/reference/components/grid.mdx:67]:
```suggestion
        col="1 / span 4"
```

File: website/docs/reference/components/grid.mdx [hunk 3]
```diff
@@ -21,94 +28,162 @@ import LiveExampleTabs from '@site/src/components/LiveExampleTabs/LiveExampleTab
...
-  GridOptions,
-  GridItemOptions,
-} from '@pluralsight/headless-styles/types'
+import { Grid, GridItem } from '@pluralsight/react'
 ```
 
 ## Usage
 
-### Basic 2 Column Layout
+## Columns
```
Comment (Rykus0) [inline website/docs/reference/components/grid.mdx:36]:
```suggestion
### Columns
```

File: website/docs/reference/components/grid.mdx [hunk 4]
```diff
@@ -21,94 +28,162 @@ import LiveExampleTabs from '@site/src/components/LiveExampleTabs/LiveExampleTab
...
+          }}
+        />
+      ))}
+    </Grid>
   )
 }
 ```
 
-## Reference
+## Spanning columns
```
Comment (Rykus0) [inline website/docs/reference/components/grid.mdx:58]:
```suggestion
### Spanning columns
```

**Pull request merged**